src/DssSpell.t.base.sol

```diff
@@ -2370,4 +2540,234 @@ contract DssSpellTestBase is Config, DssTest {
         // Dump all dai for next run
         vat.move(address(this), address(0x0), vat.dai(address(this)));
     }
+
+    function _testSplitter() internal {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        assertEq(vow.flapper(), address(split), "TestError/invalid-vow-flapper");
+        assertEq(split.flapper(), address(flap), "TestError/invalid-split-flapper");
+        assertEq(flap.gem(), address(sky), "TestError/invalid-flapper-gem");
+        assertEq(flap.pip(), addr.addr("FLAP_SKY_ORACLE"), "TestError/invalid-flapper-pip");
+        assertEq(flap.pair(), addr.addr("UNIV2USDSSKY"), "TestError/invalid-flapper-pair");
+
+        // Check splitter and flapper
+        {
+            // Leave surplus buffer ready to be flapped
+            vow.heal(vat.sin(address(vow)) - (vow.Sin() + vow.Ash()));
+            // Ensure flapping is possible
+            stdstore
+                .target(address(vat))
+                .sig("dai(address)")
+                .with_key(address(vow))
+                .checked_write(vat.sin(address(vow)) + vow.bump() + vow.hump());
+
+            GemAbstract pair = GemAbstract(addr.addr("UNIV2USDSSKY"));
+            FlapOracleLike pip = FlapOracleLike(flap.pip());
+
+            vm.prank(address(flap));
+            uint256 price = uint256(pip.read());
+
+            // Ensure there is enough liquidity
+            uint256 usdsWad = 150_000_000 * WAD;
+            GodMode.setBalance(address(usds), address(pair), usdsWad);
+            // Ensure price is within the tolerane (flap.want() + delta (1 p.p.))
+            uint256 skyWad = usdsWad * (flap.want() + 10**16) / price;
+            GodMode.setBalance(address(sky), address(pair), skyWad);
```

Comment:
Shouldn't there already be enough liquidity after the spell execution, since we've migrated full DAI/MKR liquidity into this pool? Otherwise, this might be an actual problem (SBE failing, leading to potential depeg)

Let's remove `GodMode.setBalance` calls to overwrite pool balance for those tests to make practical sense

Reply:
Since this is a base test, it means that it is going to be executed every time. It would be bad if it fails when for the liquidity in the pool is low for some reason.

In order to avoid flaky tests, I believe it would be better to leave the test as it is, because the objective is to assert that the Flapper works as expected when the state of the system allows it to.

If the liquidity is too low, it is known that the Flapper will fail, so having it actually failing would bring no additional benefit.

Reply:
Well, the point is to test two different things (univ2 pool migration and new splitter + flapper) together, end-to-end with as little assumptions as possible. The point about the test being potentially "flaky" is weak. What is the probability of the new pool to be empty after we've moved liquidity there? Very low. Even in case of "base test" logic. Why base test can not enforce that enough liquidity is kept in the pool to avoid unexpected SBE failure?

Reply:
> Well, the point is to test two different things (univ2 pool migration and new splitter + flapper) together [...]

I am not entirely sure that this is needed. We can test both of them separately. They are pretty much orthogonal, as long as we're using the same `Pair`.
Also, the migration is a 1-off, while the Flapper will remain active. I believe it is better to not conflate both.

> why base test can not enforce that enough liquidity is kept in the pool to avoid unexpected SBE failure?

This part of the system is not under control of spells, the state can change at any moment. This test would only "enforce" anything if we are in the middle of a spell process and someone actually runs it. The community would probably notice the problem way before that.

There is however one scenario that comes to my mind: in case we ever need to replace the pool (i.e.: using a UniswapV3 Pair instead), this test would still pass even if the crafter forgets to update the reference after the migration happens. However, I believe this could be caught in the review process.

If it's not a blocker, I'd rather keep the test as is.